src/request.rs

```diff
@@ -1,220 +1,378 @@
+                let url = add_query_parameters(url, &query)?;
+
+                builder
+                    .method(HttpMethod::PUT)
+                    .uri(url)
+                    .header(header::CONTENT_TYPE, content_type)
+                    .body(AsyncBody::from_reader(body))
+                    .map_err(|_| crate::errors::Error::InvalidRequest)?
+                    .send_async()
+                    .await?
+            }
+        };
 
-    let mut response = match method {
-        Method::Get { query } => {
-            let url = add_query_parameters(url, &query)?;
-
-            builder
-                .method(HttpMethod::GET)
-                .uri(url)
-                .body(())
-                .map_err(|_| crate::errors::Error::InvalidRequest)?
-                .send_async()
-                .await?
-        }
-        Method::Delete { query } => {
-            let url = add_query_parameters(url, &query)?;
-
-            builder
-                .method(HttpMethod::DELETE)
-                .uri(url)
-                .body(())
-                .map_err(|_| crate::errors::Error::InvalidRequest)?
-                .send_async()
-                .await?
-        }
-        Method::Post { query, body } => {
-            let url = add_query_parameters(url, &query)?;
-
-            builder
-                .method(HttpMethod::POST)
-                .uri(url)
-                .header(header::CONTENT_TYPE, content_type)
-                .body(AsyncBody::from_reader(body))
-                .map_err(|_| crate::errors::Error::InvalidRequest)?
-                .send_async()
-                .await?
-        }
-        Method::Patch { query, body } => {
-            let url = add_query_parameters(url, &query)?;
-
-            builder
-                .method(HttpMethod::PATCH)
-                .uri(url)
-                .header(header::CONTENT_TYPE, content_type)
-                .body(AsyncBody::from_reader(body))
-                .map_err(|_| crate::errors::Error::InvalidRequest)?
-                .send_async()
-                .await?
+        let status = response.status().as_u16();
+
+        let mut body = response
+            .text()
+            .await
+            .map_err(|e| crate::errors::Error::HttpError(e.into()))?;
+
+        if body.is_empty() {
+            body = "null".to_string();
         }
-        Method::Put { query, body } => {
-            let url = add_query_parameters(url, &query)?;
-
-            builder
-                .method(HttpMethod::PUT)
-                .uri(url)
-                .header(header::CONTENT_TYPE, content_type)
-                .body(AsyncBody::from_reader(body))
-                .map_err(|_| crate::errors::Error::InvalidRequest)?
-                .send_async()
-                .await?
+
+        parse_response(status, expected_status_code, body)
+    }
+}
+
+#[cfg(feature = "isahc")]
+#[cfg(target_arch = "wasm32")]
+#[async_trait(?Send)]
+impl HttpClient for IsahcClient {
```

Comment:
not sure this work

Reply:
Fixed to use `WebSysClient` for wasm.

https://github.com/meilisearch/meilisearch-rust/commit/71f2abebb91da8e9b3afe29745730804f7f869ec